src/Nix/Exec.hs

```diff
@@ -428,7 +429,7 @@ coerceToString = \case
     NVConstant (NUri u)   -> pure $ show u
     NVConstant NNull      -> pure ""
 
-    NVStr t _ -> pure $ Text.unpack t
+    NVStr ns -> pure $ Text.unpack $ stringIntentionallyDropContext ns
```

Comment:
Not sure where `coerceToString` is being used, but if it's inspired by the function of the same name in the C++ codebase we definitely don't want to drop context here.

Reply:
It's meant to become an exact replica of the C++ version.